bite-size-notes/week-2-d-Intro-To-Convex-Optimization.md

```diff
@@ -0,0 +1,143 @@
+
+In controls, optimization allows us not just to choose a feasible control input for our system, or even just a good control for our system, but to actually find the best control input for our system based on our constraints, desired behavior, and current state. This can be a powerful tool. However, there are some limiting factors. While the mathematical framework for setting up an optimization problem can be used to define most problems, actually solving a problem can yield very crude solutions. The key to solving an optimization problem simply and convincingly is to frame the problem using a convex objective function and convex subset[3]. The rest of these notes will explain what convex functions, subsets, and optimizations are and how to recognize and build systems that fit within them.   
+
+
+## Preliminaries  
+
+### Convexity 
+Mathematics defines a function as convex if a line segment can be drawn between any two distinct points, and the rest of the function stays below that line.  
+
+<p align="center">
+  <img src="figs/Convex_vs._Not-convex.jpg" alt="drawing" width="500"/>
+</p> 
+[4]
+
+The function is usually visually shaped like a bowl or a U. 
+
+The significance of convexity in optimization is primarily due to the property of convex functions to have a single global minimum. Since a Convex function is bowl-shaped, everything is sloped to a single lowest point. This means there is only one optimal solution, and all paths lead towards it. You can imagine that for a function with many dips and valleys, it may be challenging to find the global minimum since there are so many local minimums. 
+
+**Convex Set:** Contains a line segment between any two points in the set.  
+
+ex:   
+<p align="center">
+  <img src="figs/ConvexExamples.jpg" alt="drawing" width="500"/>
+</p>  
+[1]
+The first shape is a convex set, while the other two are not convex.  
+
+Other types of convex sets:
+
+* Convex Hull  
+* Convex Cone  
+* Hyperplanes and Halfspaces    
+* Euclidian balls and Ellipsoids  
+* Norm balls and norm cones  
+* Polyhedra  
+* Positive semidefinite cone
+
+For more information on the above convex sets, take a look at [1]
+
+**Convex Functions:**  
+* $f: \pmb{R_n} \rightarrow \pmb{R}$ is convex if the domain of $f$ is a convex set and for all $x,y \in f, 0 \leq \theta\leq1$
+  
+<p align="center">
+  <img src="figs/ConvexFunc.jpg" alt="drawing" width="500"/>
+</p>   
+[1]  
+
+* $f$ is concave if -f is convex
+* $f$ is strictly convex if the domain of $f$ is convex and for $x,y \in$ domain of $f, x \neq, 0 <\theta< 1,$ 
+
+Example of convex functions: 
+
+* affine: ax+b on **R** for any $a,b \in \pmb{R}$
+* exponential: $e^{ax}$ for any $a \in \pmb{R}$
+* powers: $x^a$ on $\pmb{R_{++}}$ for $\alpha \geq 1,  \alpha \leq 0$
+* powers of absolute value: $|x|^p$ on $\pmb{R}$ for $p \geq 1$
+* positive part (relu): max {0,x} 
+
+## Main Body 
+
+### Standard form of an optimization problem 
+minimize x  in  $$f_0 (x)$$    
```

Comment:
1. add 'min_x' to the actual math equation. Do this by adding the following: 
$$
\begin{aligned}
\text{min}_x f_0 (x) 
\end{aligned}
$$

2. Do the same for subject to part.